timon/plugins/trio.py

```diff
@@ -43,29 +45,10 @@ def after_run(self):
         print("!!! run finished")
 
 
-async def trio_count():
-    global val
-    await trio.sleep(1)
-    print("triocount", val)
-    val += 1
-
-
-async def async_main_wrapper(options, cfg, run_once, t00, run_func):
-    print("R-once", run_once)
-    async with trio_asyncio.open_loop() as loop:
-        await trio_count()
-        rslt = await loop.run_asyncio(run_func, options, cfg, run_once, t00)
-        await trio_count()
-    return rslt
-
-
-def run(options, cfg, run_once, t00, run_func):
-    print("running trio mode")
+def run(func, *args, **kwargs):
+    print("running")
     instruments = []
-    if cfg.get_plugin_param('trio.instrumenting', False):
-        instruments.append(Tracer())
-    rslt = trio.run(
-        async_main_wrapper, options, cfg, run_once, t00, run_func,
-        instruments=instruments,
-        )
-    dly, rslt_loop, notifiers = rslt
+    # if cfg.get_plugin_param('trio.instrumenting', True):
```

Comment:
Why is this commented?

it should perhaps no more look in plugin params, but at ordinary params, but it should be there.

The default value if no param is found in the config should probably be False.

additionally to get-plugin_param, one might add a commodity function like.

```
    def get_param(self, name, default=None):
        return minibelt.get(
            self.cfg, *(name.split('.')), default=default)
```

or directly hardcode the get.
```
   if minibelt.get(cfg, "trio", "instrumenting", default=False):
```

Reply:
done